<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marie Saitou" />


<title>Bio326.genome</title>

<script src="site_libs/header-attrs-2.5/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bio326</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mariesaitou/Bio326">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Bio326.genome</h1>
<h4 class="author">Marie Saitou</h4>
<h4 class="date">1/28/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 1 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Bio326/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommariesaitouBio326treee424b04258656756e17017b45151a28fd764373ctargetblanke424b04a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mariesaitou/Bio326/tree/e424b04258656756e17017b45151a28fd764373c" target="_blank">e424b04</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommariesaitouBio326treee424b04258656756e17017b45151a28fd764373ctargetblanke424b04a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mariesaitou/Bio326/tree/e424b04258656756e17017b45151a28fd764373c" target="_blank">e424b04</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store

Untracked files:
    Untracked:  BIO326-RNAseq.pptx
    Untracked:  BIO326-genome.pptx
    Untracked:  Bio326.genome.html
    Untracked:  RNAseq.Rplot.pdf
    Untracked:  workflowR.bio326.R
    Untracked:  ~$BIO326-genome.pptx

Unstaged changes:
    Modified:   analysis/Bio326.genome.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Bio326.genome.Rmd</code>) and HTML (<code>docs/Bio326.genome.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/e424b04258656756e17017b45151a28fd764373c/docs/Bio326.genome.html" target="_blank">e424b04</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-02-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/aebc2f1871caf5f09fee48742cac0b404ec24e12/docs/Bio326.genome.html" target="_blank">aebc2f1</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-02-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/5ff0019d4d26539e1d150c91b570e8f4a6e562a4/docs/Bio326.genome.html" target="_blank">5ff0019</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-02-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/4fab7ae7c05eb161cda2159b633a8485373a32f1/analysis/Bio326.genome.Rmd" target="_blank">4fab7ae</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-02-01
</td>
<td>
image data set
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/4fab7ae7c05eb161cda2159b633a8485373a32f1/docs/Bio326.genome.html" target="_blank">4fab7ae</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-02-01
</td>
<td>
image data set
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/b42559418a8c15aa7d8006b60d6f313d424ed5a8/docs/Bio326.genome.html" target="_blank">b425594</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/f447408f9507fb183dfa2c589a8d062e0f772d97/analysis/Bio326.genome.Rmd" target="_blank">f447408</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-01-28
</td>
<td>
Publish the initial files for BIO326
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/067284ec8f125f22a1a9226089f2262be576ccb2/docs/Bio326.genome.html" target="_blank">067284e</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/76be30310a478d8037735fe82d4f90da8d7e3a68/analysis/Bio326.genome.Rmd" target="_blank">76be303</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-01-28
</td>
<td>
Publish the initial files for BIO326
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/1fa796da55598142b00256654f15f0319271b6c5/docs/Bio326.genome.html" target="_blank">1fa796d</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/5f3854235270e605d06126f320afd5c04ffbb7c2/docs/Bio326.genome.html" target="_blank">5f38542</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/21024428e861aab6d160a1732731b14f3b66bc16/docs/Bio326.genome.html" target="_blank">2102442</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/ae3dfb5e9c0b24d8d71fc7a9fb0ce311f9ff4493/docs/Bio326.genome.html" target="_blank">ae3dfb5</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/2e5f89a809d3bdda636f02d4f26b6b3441196493/docs/Bio326.genome.html" target="_blank">2e5f89a</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/a81d12eb4b9260e3d37add45fe6b5413cf9ad63b/docs/Bio326.genome.html" target="_blank">a81d12e</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/ab8165a49ed086870ddccc43e90c0eca99c952c4/analysis/Bio326.genome.Rmd" target="_blank">ab8165a</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-01-28
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/Bio326.genome.Rmd”))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This workflow is based on Galaxy Training materials aimed for the two-hour courses in <a href="https://www.nmbu.no/course/BIO326">Bio326</a>, NMBU 2021. Lecture slide is here: (contains hands-on materials on this page)<a href="assets/BIO326-genome/BIO326-genome.pdf">pdf</a></p>
<p>The workflow is based on the following materials but modified and updated for Galaxy.no 2021.</p>
<p><a href="https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">Quality Control</a></p>
<p><a href="https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">Mapping</a></p>
<p><a href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html">Variant Analysis</a></p>
<div id="goal-of-this-workflow" class="section level1">
<h1>0. Goal of this workflow</h1>
<p>Objective: Compare the mitochondrial variants between mother and child (human)</p>
<p>We will learn: A.How to conduct “cleaning” of the data, B.How to map the sequence read to refeerence genome, C. How to call genetic variants.</p>
</div>
<div id="galaxy-introduction" class="section level1">
<h1>1. Galaxy introduction</h1>
<div id="register-and-login-to-galaxy" class="section level2">
<h2>1-1. Register and login to Galaxy</h2>
<p>Go to <a href="https://usegalaxy.no/" class="uri">https://usegalaxy.no/</a> or <a href="https://usegalaxy.org/" class="uri">https://usegalaxy.org/</a> Galaxy is a web platform with various software for genome analyses. You should be able to log in with “Feide” information at galaxy.no (NMBU ID and password)</p>
</div>
<div id="play-around-with-galaxy" class="section level2">
<h2>1-2. Play around with Galaxy</h2>
<p><img src="assets/BIO326-genome/Slide18.png" /></p>
<p><img src="assets/BIO326-genome/Slide19.png" /> <img src="assets/BIO326-genome/Slide20.png" /></p>
<p><img src="assets/BIO326-genome/Slide21.png" /></p>
</div>
<div id="review-the-genome-analysis-workflow" class="section level2">
<h2>1-3. Review the genome analysis workflow</h2>
<p><img src="assets/BIO326-genome/Slide15.png" /></p>
</div>
</div>
<div id="quality-control" class="section level1">
<h1>2. Quality Control</h1>
<div id="introduction" class="section level2">
<h2>2-0. Introduction</h2>
<p>During sequencing, the nucleotide bases in a DNA or RNA sample (library) are determined by the sequencer. For each fragment in the library, a short sequence (=read) is generated,.</p>
<p>Modern sequencing technologies can generate a massive number of sequence reads in a single experiment. However, each instrument will generate different types and amount of errors, such as incorrect nucleotides being called. These wrongly called bases are due to the technical limitations.</p>
<p>Therefore, it is necessary to understand, identify and exclude error-types that may impact the interpretation of downstream analysis. Sequence quality control is therefore an essential first step in your analysis. Catching errors early saves time later on.</p>
<p>Objective: Conduct quality control and trimming of sequence data from two individuals, mother and child</p>
<p>We will learn: How to conduct quality control with FastQC, how to interpret FastQC output, how to conduct trimming with fastp.</p>
</div>
<div id="prapere-the-data" class="section level2">
<h2>2-1. Prapere the data</h2>
<p>Create a new history. <img src="assets/BIO326-genome/Slide22.png" /></p>
<p>Import the data set.</p>
<p><img src="assets/BIO326-genome/Slide24.png" /> Copy and paste the following URLs to import the data:</p>
<p><a href="https://zenodo.org/record/1251112/files/raw_child-ds-1.fq" class="uri">https://zenodo.org/record/1251112/files/raw_child-ds-1.fq</a></p>
<p><a href="https://zenodo.org/record/1251112/files/raw_child-ds-2.fq" class="uri">https://zenodo.org/record/1251112/files/raw_child-ds-2.fq</a></p>
<p><a href="https://zenodo.org/record/1251112/files/raw_mother-ds-1.fq" class="uri">https://zenodo.org/record/1251112/files/raw_mother-ds-1.fq</a></p>
<p><a href="https://zenodo.org/record/1251112/files/raw_mother-ds-2.fq" class="uri">https://zenodo.org/record/1251112/files/raw_mother-ds-2.fq</a></p>
<p>Rename the data for to manage them. <img src="assets/BIO326-genome/Slide25.png" /></p>
<p>Let’s have a look… <img src="assets/BIO326-genome/Slide26.png" /></p>
<p>What is written in the FastQ files: <img src="assets/BIO326-genome/Slide27.png" /> Quality scores are recorded as “ASCII” symbol. The code is different in each platform. For example, “B” is quality score “32”, which means more than 30 = “Probability of incorrect base call” is less than “1 in 1000” (accuracy &gt; 99.9%).</p>
<p><img src="assets/BIO326-genome/Slide28.png" /></p>
</div>
<div id="quality-control-with-fastqc" class="section level2">
<h2>2-2. Quality control with fastQC</h2>
<p><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastQC</a> (go to the website for details)</p>
<p><img src="assets/BIO326-genome/Slide29.png" /></p>
<p>Let’s see the result of quality check. <img src="assets/BIO326-genome/Slide30.png" /> Example results “Per base sequence quality”: X-axis is the base position, Y-axis is the read quality. Green is good, yellow is okay, red is not good. <img src="assets/BIO326-genome/Slide31.png" /></p>
<p>Our results: We see that the ending part ot our reads shows not good quality. -&gt; Trimming. <img src="assets/BIO326-genome/Slide32.png" /> Per sequence quality score. X-axis is the quality score, Y-axis is the observed count. We see that most of the reads show good quality, but still there are some low-quality reads.</p>
<p><img src="assets/BIO326-genome/Slide33.png" /></p>
<p>Sequence duplication levels. There are some duplication (small peak) due to enrichment bias. These read will be removed in the further step. <img src="assets/BIO326-genome/Slide34.png" /></p>
</div>
<div id="trimming-with-fastp" class="section level2">
<h2>2-3. Trimming with fastp</h2>
<p><a href="https://github.com/OpenGene/fastp">fastp</a> (go to the website for details)</p>
<p><img src="assets/BIO326-genome/Slide35.png" /> Let’s see how many low-quality reads are removed: <img src="assets/BIO326-genome/Slide36.png" /></p>
</div>
</div>
<div id="mapping" class="section level1">
<h1>3. Mapping</h1>
<p>Now, we are going to map reads to the reference genome (human genome) FastQ format -&gt; BAM format</p>
<p><img src="assets/BIO326-genome/Slide37.png" /></p>
<div id="mapping-with-bwa-mem" class="section level2">
<h2>3-1. Mapping with BWA-MEM</h2>
<p><img src="assets/BIO326-genome/Slide38.png" /> Taking too long? Here are the bam files: <a href="assets/BIO326-genome/mother.bam">mother</a> and <a href="assets/BIO326-genome/child.bam">child</a></p>
</div>
<div id="merge-the-two-bam-files" class="section level2">
<h2>3-2. Merge the two BAM files</h2>
<p>We will merge the mother and child BAM files into one file so that we can handle it easily. <img src="assets/BIO326-genome/Slide39.png" /></p>
</div>
<div id="remove-the-pcr-duplicates-review-2-2." class="section level2">
<h2>3-3. Remove the PCR duplicates (review 2-2.)</h2>
<p><img src="assets/BIO326-genome/Slide40.png" /></p>
</div>
<div id="left-aligning-insertiondeletion-variants" class="section level2">
<h2>3-4. Left-aligning insertion/deletion variants</h2>
<p><img src="assets/BIO326-genome/Slide42.png" /> he same insertion/deletion can often be placed at multiple positions. To prevent this, we will conduct “Left-aligning” of insertion/deletion variants with "BamLeftAlign. <img src="assets/BIO326-genome/Slide43.png" /></p>
</div>
<div id="bam-file-filtering" class="section level2">
<h2>3-5. BAM file filtering:</h2>
<p><img src="assets/BIO326-genome/Slide44.png" /> We will filter the bam file with quality and genomic region. <img src="assets/BIO326-genome/Slide45.png" /></p>
</div>
</div>
<div id="varitnt-calling" class="section level1">
<h1>4. Varitnt Calling</h1>
<div id="calling-variants-with-freebayes" class="section level2">
<h2>4-1. Calling variants with FreeBayes</h2>
<p>Let’s call variants from the BAM files and examine variants. BAM format -&gt; VCF (variant fall format)</p>
<p><img src="assets/BIO326-genome/Slide46.png" /> <img src="assets/BIO326-genome/Slide47.png" /></p>
<p><img src="assets/BIO326-genome/Slide48.png" /></p>
<p><img src="assets/BIO326-genome/Slide49.png" /></p>
<p>Taking too long? Here is the vcf file: <a href="assets/BIO326-genome/FreeBayes_mother_child.vcf">VCF</a></p>
</div>
<div id="filtering-variants" class="section level2">
<h2>4-2. Filtering variants</h2>
<p>Filter the variant based on the reliability. There are various “biases” and artifacts that can happen. <img src="assets/BIO326-genome/Slide50.png" /></p>
<p><img src="assets/BIO326-genome/Slide51.png" /></p>
<p><img src="assets/BIO326-genome/Slide52.png" /></p>
</div>
<div id="reformatting-the-vcf-file" class="section level2">
<h2>4-3. Reformatting the VCF file</h2>
<p>Just reformat VCF file so that human eye can easily read it. <img src="assets/BIO326-genome/Slide53.png" /> <img src="assets/BIO326-genome/Slide54.png" /></p>
</div>
</div>
<div id="examine-the-variants." class="section level1">
<h1>5. Examine the variants.</h1>
<p>We see two variants at position 3243 and 5539. Both are (Reference: A and Alternative:G) <img src="assets/BIO326-genome/Slide55.png" /> Let’s take a close look on column five and six. <img src="assets/BIO326-genome/Slide56.png" /> <img src="assets/BIO326-genome/Slide57.png" /> Q: Allele frequency of variants in Mother and Child</p>
<p>Posision 3243 Child : G = 0.67 Mother : G = ???</p>
<p>Posision 5538 Child : G = ??? Mother : G = ???</p>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
