<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marie Saitou" />

<meta name="date" content="2023-01-30" />

<title>Bio326_2023</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bio326</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mariesaitou/Bio326">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Bio326_2023</h1>
<h4 class="author">Marie Saitou</h4>
<h4 class="date">2023-01-30</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Bio326/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210128code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20210128)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210128code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210128)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommariesaitouBio326tree41e8c92aeb3e9e7f7b6d4ca8670867730d20eb9etargetblank41e8c92a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mariesaitou/Bio326/tree/41e8c92aeb3e9e7f7b6d4ca8670867730d20eb9e" target="_blank">41e8c92</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommariesaitouBio326tree41e8c92aeb3e9e7f7b6d4ca8670867730d20eb9etargetblank41e8c92a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mariesaitou/Bio326/tree/41e8c92aeb3e9e7f7b6d4ca8670867730d20eb9e" target="_blank">41e8c92</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Bio326_2023.Rmd</code>) and HTML
(<code>docs/Bio326_2023.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/41e8c92aeb3e9e7f7b6d4ca8670867730d20eb9e/analysis/Bio326_2023.Rmd" target="_blank">41e8c92</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-03-06
</td>
<td>
wflow_publish(c("analysis/Bio326_2023.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/ef86291db36512bb5f67a0e9705dccdf0f9ad78f/docs/Bio326_2023.html" target="_blank">ef86291</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/4082f836b739b8bd8e58e579f6ca3199d0119ed3/analysis/Bio326_2023.Rmd" target="_blank">4082f83</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-03-03
</td>
<td>
wflow_publish(c("analysis/Bio326_2023.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/b563dcbb3bbd6c8173c4d3448266460305b53418/docs/Bio326_2023.html" target="_blank">b563dcb</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/65211fd3da69a1e5ecbb882056526aa6e995c803/analysis/Bio326_2023.Rmd" target="_blank">65211fd</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-03-03
</td>
<td>
image data set
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/10c4a0a09c4468a584271c397d08c3dc6d2652fc/docs/Bio326_2023.html" target="_blank">10c4a0a</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/eb2f5bdd946ba2f70f80bf0ff6e1c0e9a8ee96b8/docs/Bio326_2023.html" target="_blank">eb2f5bd</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/748a76df5551ef2d78b8d638ef1ca5481516aa08/docs/Bio326_2023.html" target="_blank">748a76d</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/1dc72fd110db0c3c6f1b82fed5812546a9766c91/analysis/Bio326_2023.Rmd" target="_blank">1dc72fd</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-03-02
</td>
<td>
wflow_publish(c("analysis/Bio326_2023.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/91c30c1301cfd2d49187151877133f1068c616b5/docs/Bio326_2023.html" target="_blank">91c30c1</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/2016a67867d18c249ed1e077e8dd3760edcf54df/analysis/Bio326_2023.Rmd" target="_blank">2016a67</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-02-28
</td>
<td>
wflow_publish(c("analysis/Bio326_2023.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/7c8df25aec27d46c03c87a4a212b0188903bea4a/docs/Bio326_2023.html" target="_blank">7c8df25</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/8b730e0b8199176cedc7c5968cb0f72d5195e59d/docs/Bio326_2023.html" target="_blank">8b730e0</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/1d9c035fc4878bd2b1adc435b2532a373f9c2dd1/docs/Bio326_2023.html" target="_blank">1d9c035</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/f5fe8f700ecc4fbafcfd1971847f46d64c9651d2/analysis/Bio326_2023.Rmd" target="_blank">f5fe8f7</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-02-27
</td>
<td>
wflow_publish(c("analysis//Bio326_2023.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/942fc184be7690b3b78a9c1bddf50c50dd3887e8/docs/Bio326_2023.html" target="_blank">942fc18</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/5ccee4b2e2ad28881043e350442665d741dfb286/analysis/Bio326_2023.Rmd" target="_blank">5ccee4b</a>
</td>
<td>
mariesaitou
</td>
<td>
2023-02-27
</td>
<td>
wflow_publish(c("analysis//Bio326_2023.Rmd"))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overview-of-the-tutorial" class="section level2">
<h2>Overview of the tutorial</h2>
<p>Last time, you started sequencing of cattle genome using Nanopore
MinION. For the following three sessions we will learn:</p>
<ul>
<li><p>How to use Orion and conduct genome analysis</p></li>
<li><p>Quality check, Read filtering, mapping to the reference genome
and variant calling</p></li>
<li><p>How to interpret summary statistics of Nanopore sequence
data</p></li>
<li><p>How to interpret variant data</p></li>
</ul>
<p>In this tutorial, we will investigate a small subset of bull genome
sequence.</p>
<p>You will have an access to the whole datasets later for your reports
as soon as the computation is done.</p>
<p>Also, Matthew and I prepared some questions in each section.</p>
<p>Please discuss and try the quizzes to deepen your understandig on
Nanopore data.</p>
<p><a
href="https://github.com/TheMEMOLab/Bio326-NMBU/blob/main/Doc/BestPracticesOrionHPC.md">Review
the previous practice</a></p>
</div>
<div id="day1" class="section level1">
<h1>DAY1</h1>
<p>Overview.</p>
<p>Quality check -&gt; Trimming of low quality reads -&gt; Quality
check</p>
<div id="connect-to-orion-and-the-preparation" class="section level2">
<h2>Connect to Orion and the preparation</h2>
<p>Go to <a href="https://orion.nmbu.no/"
class="uri">https://orion.nmbu.no/</a> at NMBU or with VPN. <img
src="assets/Bio326-2022/prep1.png" style="width:80.0%" /></p>
<p>In the Terminal/Command prompt, go to your directory. <a
href="https://github.com/TheMEMOLab/Bio326-NMBU/blob/main/Doc/BestPracticesOrionHPC.md#temporary-working-directory-faster-and-more-efficient-jobs">Review:
the concept of current directry</a></p>
<pre class="bash"><code>cd your_directory</code></pre>
<p>Let’s make a directory for analysis and enter in it.</p>
<pre class="bash"><code>mkdir bull_analysis # make directory &quot;bull_analysis&quot;
cd bull_analysis # set the current directory &quot;bull_analysis&quot;
</code></pre>
<p>Now, you will inspect the fastq file, which contains Nanopore read
information.</p>
</div>
<div id="check-the-read-quality-by-nanoplot" class="section level2">
<h2>Check the read quality by Nanoplot</h2>
<div id="browse-the-inside-of-the-read-fastq-file"
class="section level3">
<h3>Browse the inside of the read (fastq) file</h3>
<p><a
href="https://github.com/TheMEMOLab/Bio326-NMBU/blob/main/Doc/BestPracticesOrionHPC.md#conda-envrionment">Review:
look into a file content in a command line</a></p>
<pre class="bash"><code>zcat /net/fs-2/scale/OrionStore/Courses/BIO326/EUK/bull_analysis/demo_data/bull_demodata_fastq.gz | more
</code></pre>
<p>How a fastq file looks.</p>
<p>Each entry in a FASTQ files consists of 4 lines:</p>
<ol style="list-style-type: decimal">
<li>A sequence identifier with information about the sequencing run.
(run time, run ID, cflow cell id … )</li>
</ol>
<p>2.The sequence (the base calls; A, C, T, G and N).</p>
<ol start="3" style="list-style-type: decimal">
<li><p>A separator, which is simply a plus (+) sign.</p></li>
<li><p>The base call quality scores. These are Phred +33 encoded, using
ASCII characters to represent the numerical quality scores.” <a
href="https://learn.gencore.bio.nyu.edu/ngs-file-formats/quality-scores/">quality
score sheet</a></p></li>
</ol>
<ul>
<li>by <a
href="https://knowledge.illumina.com/software/general/software-general-reference_material-list/000002211">Illumina
Knowledge</a></li>
</ul>
<p><img src="assets/Bio326-2023/image1.png" style="width:80.0%" /></p>
</div>
<div id="get-basic-stats-of-the-fastq-file" class="section level3">
<h3>Get basic stats of the fastq file</h3>
<p>“zcat”-&gt; look inside</p>
<p>“wc” -&gt; word count</p>
<p>“-l” -&gt; line</p>
<pre class="bash"><code>zcat /net/fs-2/scale/OrionStore/Courses/BIO326/EUK/bull_analysis/demo_data/bull_demodata_fastq.gz | wc -l</code></pre>
<p><img src="assets/Bio326-2023/image2.png" style="width:80.0%" /></p>
</div>
</div>
<div id="discussion-point" class="section level2">
<h2><strong>Discussion Point</strong></h2>
<p>Now you got the number of lines in the fastq file.</p>
<p>How many sequence reads are in the fastq file?</p>
<details>
<summary>
<strong>Need Help?</strong>
</summary>
<p>We see that there are 96000 lines in the fastq file.</p>
<p>As we learned that “each entry in a FASTQ files consists of 4 lines”,
one read is corresponding to four lines. So in this file we have 96000/4
= 24000 reads.</p>
</details>
<div id="run-nanoplot" class="section level3">
<h3>Run Nanoplot</h3>
<p>The original fastq files may contain low quality reads. In this step,
we will use “Nanoplot” to see the quality and lentgh of each read.</p>
<p>“Singularity” is a toolset on Orion to execute software. A variety of
different bioinformatics tools are available in Singularity.</p>
<p>Make a slurm script like below and run it.</p>
<p><a href="to%20be%20updated">Review: make a slurm script</a></p>
<p><a href="https://orion.nmbu.no/en/SLURM">Review: run a slurm script
by sbatch sbatch</a></p>
<pre class="bash"><code>

#!/bin/bash
#SBATCH --job-name=Nanoplot  # sensible name for the job
#SBATCH --mail-user=yourname@nmbu.no # Email me when job is done.
#SBATCH --mem=12G 
#SBATCH --ntasks=1   
#SBATCH --cpus-per-task=8
#SBATCH --mail-type=END


singularity exec /cvmfs/singularity.galaxyproject.org/all/nanoplot:1.41.0--pyhdfd78af_0 NanoPlot -t 8  --fastq /net/fs-2/scale/OrionStore/Courses/BIO326/EUK/bull_analysis/demo_data/bull_demodata_fastq.gz --plots dot   --no_supplementary --no_static --N50 -p before
</code></pre>
<p>Nanoplot will generate the result files, named “before”xxx. Lets look
into them…</p>
<p><a href="https://orion.nmbu.no/en/connectingtoorion">Review: File
transfer between Orion and your computer</a></p>
<pre class="bash"><code>
# taking too long?
qlogin 

cp /net/fs-2/scale/OrionStore/Courses/BIO326/EUK/bull_analysis/demo_data/beforeNanoPlot-report.html beforeNanoPlot-report.html
</code></pre>
<p>Open “beforeNanoPlot-report.html” on your local computer</p>
<p><img src="assets/Bio326-2023/image3.png" style="width:80.0%" /></p>
<p><img src="assets/Bio326-2023/image4.png" style="width:80.0%" /></p>
<p><img src="assets/Bio326-2023/image5.png" style="width:80.0%" /></p>
</div>
</div>
<div id="the-end-of-day1.-well-done" class="section level2">
<h2>The end of Day1. Well done!</h2>
</div>
</div>
<div id="day2" class="section level1">
<h1>DAY2</h1>
<p>Everything you need in case scripts do not work well</p>
<pre class="bash"><code>ls /net/fs-2/scale/OrionStore/Courses/BIO326/EUK/bull_analysis/demo_data

# use cp command to copy files

# or run the full slurm script
sbatch /net/fs-2/scale/OrionStore/Courses/BIO326/EUK/bull_analysis/demo_data/Bio326_2023_full.slurm</code></pre>
<ul>
<li><p>Filter low quality reads and short reads</p></li>
<li><p>Map the reads to the reference genome</p></li>
<li><p>Detect variants</p></li>
</ul>
<p><img src="assets/BIO326-genome/Slide14.png"
style="width:80.0%" /></p>
<div id="filtering-by-nanofilt" class="section level2">
<h2>Filtering by Nanofilt</h2>
<pre class="bash"><code>
#!/bin/bash
#SBATCH --job-name=Nanoplot  # sensible name for the job
#SBATCH --mail-user=yourname@nmbu.no # Email me when job is done.
#SBATCH --mem=12G 
#SBATCH --ntasks=1   
#SBATCH --mail-type=END

gunzip -c /net/fs-2/scale/OrionStore/Courses/BIO326/EUK/bull_analysis/demo_data/bull_demodata_fastq.gz | singularity exec  /cvmfs/singularity.galaxyproject.org/all/nanofilt:2.8.0--py_0 NanoFilt -q 10 -l 500 | gzip &gt; cleaned.bull.fastq.gz
</code></pre>
<p>-l, Filter on a minimum read length</p>
<p>-q, Filter on a minimum average read quality score</p>
<p>In this case, we are removing reads lower than quality score 10 and
shorter than 500 bases.</p>
</div>
<div id="compare-the-before-and-after-cleaning-sequences"
class="section level2">
<h2>Compare the before and after cleaning sequences</h2>
<p>Run Nanoplot again on the cleaned sequences.</p>
<details>
<summary>
<strong>Need help?</strong>
</summary>
<pre class="bash"><code>
#!/bin/bash
#SBATCH --job-name=Nanoplot  # sensible name for the job
#SBATCH --mail-user=yourname@nmbu.no # Email me when job is done.
#SBATCH --mem=12G 
#SBATCH --ntasks=1   
#SBATCH --cpus-per-task=8
#SBATCH --mail-type=END

singularity exec /cvmfs/singularity.galaxyproject.org/all/nanoplot:1.41.0--pyhdfd78af_0 NanoPlot -t 8  --fastq cleaned.bull.fastq.gz  --N50  --no_supplementary --no_static  --plots dot   -p after
</code></pre>
</details>
<p>Open “afterNanoPlot-report.html” on your local computer.</p>
<pre class="bash"><code>
# taking too long?
qlogin 

cp /net/fs-2/scale/OrionStore/Courses/BIO326/EUK/bull_analysis/demo_data/afterNanoPlot-report.html afterNanoPlot-report.html
</code></pre>
<p><img src="assets/Bio326-2023/image6.png" style="width:80.0%" /></p>
<p><img src="assets/Bio326-2023/image7.png" style="width:80.0%" /></p>
</div>
<div id="discussion-point-1" class="section level2">
<h2><strong>Discussion Point</strong></h2>
<p>Did you see the difference of read and quality distribution between
before and after the filtering?</p>
<p>Map the reads to the reference genome -&gt; Detect variants
(difference from the reference genome)</p>
</div>
<div id="find-out-where-minimap2-is-in-singularity"
class="section level2">
<h2>Find out where “minimap2” is in Singularity</h2>
<pre class="bash"><code>qlogin
find /cvmfs/singularity.galaxyproject.org/all/  -name minimap2*  

# and you will find
# /cvmfs/singularity.galaxyproject.org/all/minimap2:2.24--h7132678_1
</code></pre>
<p>In case Singularity does not work … use <a
href="https://github.com/TheMEMOLab/Bio326-NMBU/blob/main/Doc/SoftwareManagmentWithConda.md">conda</a></p>
</div>
</div>
<div id="day3" class="section level1">
<h1>DAY3</h1>
<div id="inspect-variants" class="section level2">
<h2>Inspect variants</h2>
</div>
<div id="run-minimap-and-map-the-reads-to-the-reference-genome"
class="section level2">
<h2>run Minimap and map the reads to the reference genome</h2>
<pre class="bash"><code>

#!/bin/bash
#SBATCH --job-name=Nanoplot  # sensible name for the job
#SBATCH --mail-user=yourname@nmbu.no # Email me when job is done.
#SBATCH --mem=12G 
#SBATCH --ntasks=1   
#SBATCH --cpus-per-task=8
#SBATCH --mail-type=END

singularity exec /cvmfs/singularity.galaxyproject.org/all/minimap2:2.24--h7132678_1 minimap2 -t 8 -a -a /net/fs-2/scale/OrionStore/Courses/BIO326/EUK/bull_analysis/demo_data/Bos_taurus.fa.gz  cleaned.bull.fastq.gz &gt; bull.sam
# updated (the reference Bos taurus fasta location)


# convert the sam file to bam format
singularity exec  /cvmfs/singularity.galaxyproject.org/all/samtools:1.16.1--h6899075_1 samtools view -S -b bull.sam &gt; bull0.bam

## sort the bam file
singularity exec  /cvmfs/singularity.galaxyproject.org/all/samtools:1.16.1--h6899075_1 samtools sort bull0.bam -o bull.bam

# index the bam file
singularity exec  /cvmfs/singularity.galaxyproject.org/all/samtools:1.16.1--h6899075_1 samtools index -M  bull.bam


# Variant Calling using Sniffles
singularity exec /cvmfs/singularity.galaxyproject.org/all/sniffles:2.0.7--pyhdfd78af_0 sniffles --input  bull.bam --vcf bull.vcf
</code></pre>
<p>Error correction with Pilon/Medaka</p>
<p>(Skip this time as it takes time… You will learn the error correction
in the prokaryotic part. )</p>
<pre class="bash"><code>
# taking too long?
qlogin 
ls /net/fs-2/scale/OrionStore/Courses/BIO326/EUK/bull_analysis/demo_data/

# and copy the file you need (the final product is .vcf file)</code></pre>
<p>Now you got the variant file!</p>
<p><a href="assets/Bio326-2023/bull.vcf">resulting vcf file</a></p>
<ul>
<li>Celian will explain how to read a vcf file.</li>
</ul>
<pre class="bash"><code># INFO field

grep &#39;^##&#39; bull.vcf | tail -n 20

# variants
grep -v &#39;^##&#39; bull.vcf | more
</code></pre>
<p><img src="assets/Bio326-2023/image11.png" style="width:80.0%" /></p>
<p>Important parameters</p>
<p>1 16849578 : location of the variant</p>
<p>SVTYPE=DEL;SVLEN=-60 : size and type of the variant</p>
<p>0/1 : genotype</p>
<p>(you can open a vcf file in notepad, excel etc.)</p>
<p>Now you have variants! Lets see what genes are affected by the
variants.</p>
</div>
<div id="estimate-the-effect-of-variants" class="section level2">
<h2>Estimate the effect of variants</h2>
<p><a href="https://www.ensembl.org/Tools/VEP">Go to VEP (Variant Effect
Predictor)</a></p>
<p>Variant Effect predictor tells us where in the genome the discovered
variants are located (genic, regulartory, etc…)</p>
<p>Select “cow” as the reference species.</p>
<p><img src="assets/Bio326-2023/image14.png" style="width:80.0%" /></p>
<p>Upload: bull.vcf - downloaded from Orion or the section above as the
file to investigate.</p>
<p><img src="assets/Bio326-2023/image15.png" style="width:80.0%" /></p>
<p>There are 428 variants; 88 genes are affected by these varaints.</p>
<p>What are the most affected genes?</p>
<p>Click “Filters” and set “Impact is HIGH” to select highly impact
variants.</p>
<p><img src="assets/Bio326-2023/image16.png" style="width:80.0%" /></p>
<p>There are some frameshift/transcript ablation variants.</p>
<p>Let’s closely investigate “25:7668-11178 deletion transcript_ablation
5S_rRNA”.</p>
<p><a href="https://igv.org/app/">Go to IGV (Integrative Genomics
Viewer)</a></p>
<p>Select “Genome”: Cow (bosTau9)</p>
<p><img src="assets/Bio326-2023/image12.png" style="width:40.0%" /></p>
<p>Select “Tracks”: bull.vcf - downloaded from Orion or the section
above.</p>
<p><img src="assets/Bio326-2023/image13.png" style="width:40.0%" /></p>
<p>Add “Ensembl Genes” to the “Track” and specify the region around
chr25:7668-11178.</p>
<p>Did you find the large deletion that is covering the gene? How long
is the deletion?</p>
<p><img src="assets/Bio326-2023/image17.png" style="width:80.0%" /></p>
</div>
</div>
<div id="real-result-files-for-the-report" class="section level1">
<h1>Real result files for the report</h1>
<p>Result will be coming to the following folder</p>
<pre class="bash"><code>ls /net/fs-2/scale/OrionStore/Courses/BIO326/EUK/bull_analysis/real_data

# please use cp to copy the files to your directory</code></pre>
<p>You can also download the result files from <a
href="https://www.dropbox.com/sh/2qbvt7y4099j4n9/AACTOQdOBnVfeJkp6JIafQCUa?dl=0">here</a>.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.10      compiler_4.2.2   pillar_1.8.1     bslib_0.4.2     
 [5] later_1.3.0      git2r_0.31.0     jquerylib_0.1.4  tools_4.2.2     
 [9] getPass_0.2-2    digest_0.6.31    jsonlite_1.8.4   evaluate_0.20   
[13] lifecycle_1.0.3  tibble_3.1.8     pkgconfig_2.0.3  rlang_1.0.6     
[17] cli_3.6.0        rstudioapi_0.14  yaml_2.3.7       xfun_0.37       
[21] fastmap_1.1.0    httr_1.4.4       stringr_1.5.0    knitr_1.42      
[25] fs_1.6.1         vctrs_0.5.2      sass_0.4.5       rprojroot_2.0.3 
[29] glue_1.6.2       R6_2.5.1         processx_3.8.0   fansi_1.0.4     
[33] rmarkdown_2.20   callr_3.7.3      magrittr_2.0.3   whisker_0.4.1   
[37] ps_1.7.2         promises_1.2.0.1 htmltools_0.5.4  httpuv_1.6.8    
[41] utf8_1.2.3       stringi_1.7.12   cachem_1.0.6    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
