<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marie Saitou" />


<title>Bio326.RNAseq</title>

<script src="site_libs/header-attrs-2.5/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bio326</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mariesaitou/Bio326">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Bio326.RNAseq</h1>
<h4 class="author">Marie Saitou</h4>
<h4 class="date">2/4/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Bio326/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210128code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210128)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210128code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210128)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommariesaitouBio326tree8c5b0d66825139ff2b417f10a2790b01e191a90dtargetblank8c5b0d6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mariesaitou/Bio326/tree/8c5b0d66825139ff2b417f10a2790b01e191a90d" target="_blank">8c5b0d6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommariesaitouBio326tree8c5b0d66825139ff2b417f10a2790b01e191a90dtargetblank8c5b0d6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mariesaitou/Bio326/tree/8c5b0d66825139ff2b417f10a2790b01e191a90d" target="_blank">8c5b0d6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store

Untracked files:
    Untracked:  BIO326-RNAseq.pptx
    Untracked:  BIO326-genome/
    Untracked:  BIO326_Introduction to sequence technology and protocols_3rd_FEB_2021.pdf
    Untracked:  BIO326_Introduction to sequence technology and protocols_3rd_FEB_2021.pptx
    Untracked:  BIO326_Introduction to sequence technology and protocols_5th_FEB_2021.pptx
    Untracked:  Bio326.genome.html
    Untracked:  RNAseq.Rplot.pdf
    Untracked:  workflowR.bio326.R
    Untracked:  ~$BIO326_Introduction to sequence technology and protocols_5th_FEB_2021.pptx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Bio326.RNAseq.Rmd</code>) and HTML (<code>docs/Bio326.RNAseq.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/a3850b116a6105954798ed7f8f0037ec1ff1500f/analysis/Bio326.RNAseq.Rmd" target="_blank">a3850b1</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-02-04
</td>
<td>
image data set
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/Bio326/ad166a146d943a8334088853a6caa60e6c9075a7/docs/Bio326.RNAseq.html" target="_blank">ad166a1</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-02-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/Bio326/blob/8e170de1fdc87231b9371b2ff50815ae8ad65afb/analysis/Bio326.RNAseq.Rmd" target="_blank">8e170de</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-02-04
</td>
<td>
Add my first analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This workflow is based on Galaxy Training materials aimed for the two-hour courses in <a href="https://www.nmbu.no/course/BIO326">Bio326</a>, NMBU 2021.</p>
<p>The workflow is based on the following material but modified for Galaxy.no 2021.</p>
<p><a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html#functional-enrichment-analysis-of-the-de-genes">RNAseq</a> <a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html">gene extraction</a> <a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html">vidualization</a></p>
<p><img src="assets/BIO326-transcriptome/Slide1.png" /></p>
<div id="goal-of-this-workflow" class="section level1">
<h1>0. Goal of this workflow</h1>
<p>Objective: Compare the gene expression pattern of fruit fly with and without RNA interference treatment of Pasilla gene.</p>
<p>We will learn: A.How to prepare gene expression matrix from raw sequencing reads, B. How to conduct downstream analyses on deferentially expressed genes</p>
</div>
<div id="from-raw-reads-to-gene-expression-matrix" class="section level1">
<h1>1. From raw reads to gene expression matrix</h1>
<div id="review-the-rna-seq-analysis-workflow" class="section level2">
<h2>1-1. Review the RNA-seq analysis workflow</h2>
<p><img src="assets/BIO326-transcriptome/Slide2.png" /> ## 1-2. Get the Data</p>
<p>There are four groups and seven individuals in total.</p>
<ul>
<li>paired end sequencing, RNAi-treated</li>
<li>single end sequencing, RNAi-treated</li>
<li>paired end sequencing, not treated</li>
<li>single end sequencing, nottreated</li>
</ul>
<p>However, since each data is huge and takes long time to upload all, let’s use the PE_untreated files to learn the initial analysis. I will provide you all the processed data sets later.</p>
<p>—- PE_untreated (please use this) —- Please make a new history and import the data four files. You can always go back to the previous material to review the process. Rename the files as you like.</p>
<pre class="r"><code>https://zenodo.org/record/1185122/files/GSM461177_1.fastqsanger
https://zenodo.org/record/1185122/files/GSM461177_2.fastqsanger 
https://zenodo.org/record/1185122/files/GSM461178_1.fastqsanger
https://zenodo.org/record/1185122/files/GSM461178_2.fastqsanger</code></pre>
<p>—- PE_treated (you don;t have to use the following files for now)</p>
<pre class="r"><code>https://zenodo.org/record/1185122/files/GSM461180_1.fastqsanger
https://zenodo.org/record/1185122/files/GSM461180_2.fastqsanger
https://zenodo.org/record/1185122/files/GSM461181_1.fastqsanger
https://zenodo.org/record/1185122/files/GSM461181_2.fastqsanger</code></pre>
<p>—- SE_untreated</p>
<pre class="r"><code>https://zenodo.org/record/1185122/files/GSM461176.fastqsanger
https://zenodo.org/record/1185122/files/GSM461182.fastqsanger</code></pre>
<p>—- SE_treated</p>
<pre class="r"><code>https://zenodo.org/record/1185122/files/GSM461179.fastqsanger</code></pre>
</div>
<div id="filter-the-bad-quality-reads-with-fastp-like-the-previous-genome-analysis." class="section level2">
<h2>1-3. Filter the bad-quality reads with fastp like the previous genome analysis.</h2>
<p>Conduct fastp to each file. When your input used paired-end sequencing (you input two files), fastp will give you back two files, “Read 1 output” and “Read 2 output”.</p>
<p>OK, next step is new! We will quantify abundances of transcripts using the reference transcriptome.</p>
<p>We will use the latest (2021) fly reference transctiptome.</p>
<pre class="r"><code>assets/BIO326-transcriptome/Drosophila_melanogaster.BDGP6.28.cdna.all.fa</code></pre>
</div>
<div id="quantify-abundances-of-transcripts-with-kallisto" class="section level2">
<h2>1-4. Quantify abundances of transcripts with Kallisto</h2>
<p><a href="https://pachterlab.github.io/kallisto/manual" class="uri">https://pachterlab.github.io/kallisto/manual</a></p>
<p><img src="assets/BIO326-transcriptome/Slide3.png" /> Let’s briefly have a look on the data with the eye icon.</p>
<p><img src="assets/BIO326-transcriptome/Slide4.png" /></p>
</div>
<div id="summarize-transcripts-into-genes-with-tximport." class="section level2">
<h2>1-5. Summarize transcripts into genes with tximport.</h2>
<p>We will use a file contains gene/transcript information for the conversion:</p>
<pre class="r"><code>assets/BIO326-transcriptome/Drosophila_melanogaster.BDGP6.28.102.gff3</code></pre>
<p>Let’s briefly have a look on the gff3 file. More info: <a href="http://gmod.org/wiki/GFF3" class="uri">http://gmod.org/wiki/GFF3</a></p>
<p><img src="assets/BIO326-transcriptome/Slide5.png" /> Run tximport with the Kallisto output (two files for two individuals, for each) with the GFF3 file. <img src="assets/BIO326-transcriptome/Slide6.png" /></p>
<p>You can see gene names and gene abundance in the output file. <img src="assets/BIO326-transcriptome/Slide7.png" /></p>
</div>
</div>
<div id="differencially-expressed-gene-analysis" class="section level1">
<h1>2. Differencially expressed gene analysis</h1>
<div id="prepare-the-data" class="section level2">
<h2>2-1. Prepare the data</h2>
<p>Now, I pre-made the gene abundance table of seven individuals. Remember that there are four groups and seven individuals in total.</p>
<ul>
<li>paired end sequencing, RNAi-treated</li>
<li>single end sequencing, RNAi-treated</li>
<li>paired end sequencing, not treated</li>
<li>single end sequencing, nottreated</li>
</ul>
<pre class="r"><code>assets/BIO326-transcriptome/PE_treated_GSM461180.tsv
assets/BIO326-transcriptome/PE_treated_GSM461181.tsv
assets/BIO326-transcriptome/PE_untreated_GSM461177.tsv
assets/BIO326-transcriptome/PE_untreated_GSM461178.tsv
assets/BIO326-transcriptome/SE_untreated_GSM461182.tsv
assets/BIO326-transcriptome/SE_treated_GSM461179.tsv
assets/BIO326-transcriptome/SE_untreated_GSM461176.tsv</code></pre>
<p>You may need to make sure that the data format is “tabular”. <img src="assets/BIO326-transcriptome/Slide8.png" /></p>
</div>
<div id="deseq" class="section level2">
<h2>2-2. DeSeq</h2>
<p>Deseq runs various analyses on RNA-sequencing data. We need to input what condition we want to compare. Here, our main perpose is to know the effect of RNAi. Additionally, the sequencing platform (single-end or paired-end) may have affected.</p>
<pre class="r"><code>DESeq2  parameters:
“how”: Select datasets per level
In “1: Factor”
“Specify a factor name”: Treatment
In “1: Factor level”:
“Specify a factor level”: treated
param-files “Counts file(s)”: the 3 gene count files with treat in their name
In “2: Factor level”:
“Specify a factor level”: untreated
param-files “Counts file(s)”: the 4 gene count files with untreat in their name
Click on param-repeat “Insert Factor” (not on “Insert Factor level”)
In “2: Factor”
“Specify a factor name” to Sequencing
In “1: Factor level”:
“Specify a factor level”: PE
param-files “Counts file(s)”: the 4 gene count files with paired in their name
In “2: Factor level”:
“Specify a factor level”: SE
param-files “Counts file(s)”: the 3 gene count files with single in their name
“Files have header?”: No
“Visualising the analysis results”: Yes</code></pre>
<p><img src="assets/BIO326-transcriptome/Slide9.png" /></p>
</div>
<div id="examine-the-results." class="section level2">
<h2>2-3. Examine the results.</h2>
<p>First, take a look on the DESeq “result file”. It contains the result of differencially expressed gene analysis for the RNAi reatment. <img src="assets/BIO326-transcriptome/Slide10.png" /></p>
<p>Next, take a look on the DESeq “plots”. They will tell you the overall trend.</p>
<p>Let’s examine the transcriptome difference/smilarlity between samples/conditions.</p>
<p><img src="assets/BIO326-transcriptome/Slide11.png" /></p>
<p><img src="assets/BIO326-transcriptome/Slide12.png" /></p>
<p>What is a p-value? In statistics, the p-value is the probability of obtaining results as “extreme” as the observed results of a statistical hypothesis test, assuming that the null hypothesis is correct. We will discuss this later as well.</p>
<p><img src="assets/BIO326-transcriptome/Slide13.png" /></p>
</div>
<div id="volcano-plot" class="section level2">
<h2>2-4. Volcano plot</h2>
<p>Visualize the log2 fold change and adjusted p-value simultaniously.</p>
<p><img src="assets/BIO326-transcriptome/Slide14.png" /></p>
<p><img src="assets/BIO326-transcriptome/Slide15.png" /> ## 2-6. Examine gene functions Pick up a couple of genes with color (top differenciated)), and search for the genes in this database to investigate their known function. (top-right, junp to gene) <a href="https://flybase.org/" class="uri">https://flybase.org/</a></p>
</div>
<div id="extract-differencially-expressed-genes" class="section level2">
<h2>2-7. Extract differencially expressed genes</h2>
<p>Reformat the DESeq2 result. What we will do is very simple but the Galaxy operation may be tricky.</p>
<p>Clock the top-left “Download from URL …” icon.</p>
<p>Copy and paste “GeneID Basemean log2(FC) StdErr Wald-Stats P-value P-adj” You may need to change the datatype to tabular.</p>
<p><img src="assets/BIO326-transcriptome/Slide16.png" /></p>
<p>Concatenate the header you made above with the DESeq result.</p>
<p><img src="assets/BIO326-transcriptome/Slide17.png" /></p>
<p>Extract the genes with adjusted p-value less than 0.01. <img src="assets/BIO326-transcriptome/Slide18.png" /></p>
<p>Convert the datatype of extracted gene file to tsv. By doing so, you can download the file and examine it with Microsoft Excel etc.</p>
<p><img src="assets/BIO326-transcriptome/Slide19.png" /></p>
<p>Let’s have a look on genes in the downloaded files. How many genes did you get? Is it same as your teammates?</p>
</div>
<div id="gene-ontology-analysis" class="section level2">
<h2>2-8. Gene Ontology analysis</h2>
<p>Let’s examine the function of differencially expressed genes. Go to <a href="http://cbl-gorilla.cs.technion.ac.il/" class="uri">http://cbl-gorilla.cs.technion.ac.il/</a></p>
<p><img src="assets/BIO326-transcriptome/Slide20.png" /> Download the “background” file. (All the genes are described).</p>
<pre class="r"><code>assets/BIO326-transcriptome/GO_background_fly.csv</code></pre>
<p>Input the differencially expressed gene names and background gene file. <img src="assets/BIO326-transcriptome/Slide21.png" /></p>
<p>Explore the website and examine the potenfial function of the target gene that are interferred by RNAi.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5       rstudioapi_0.11  whisker_0.4      knitr_1.30      
 [5] magrittr_1.5     R6_2.4.1         rlang_0.4.8      stringr_1.4.0   
 [9] tools_4.0.2      xfun_0.18        git2r_0.27.1     htmltools_0.5.0 
[13] ellipsis_0.3.1   rprojroot_1.3-2  yaml_2.2.1       digest_0.6.27   
[17] tibble_3.0.4     lifecycle_0.2.0  crayon_1.3.4     later_1.1.0.1   
[21] vctrs_0.3.4      promises_1.1.1   fs_1.5.0         glue_1.4.2      
[25] evaluate_0.14    rmarkdown_2.5    stringi_1.5.3    compiler_4.0.2  
[29] pillar_1.4.6     backports_1.1.10 httpuv_1.5.4     pkgconfig_2.0.3 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
