---
title: "Bio326.genome"
author: "Marie Saitou"
date: "1/28/2021"
output: html_document
---




This workflow is based on Galaxy Training materials aimed for the two-hour courses in [Bio326](https://www.nmbu.no/course/BIO326), NMBU 2021.
Lecture slide is here: (contains hands-on materials on  this page)[pdf](assets/BIO326-genome/BIO326-genome.pdf)

The workflow is based on the following materials but modified and updated for Galaxy.no 2021. 

[Quality Control](https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html)

[Mapping](https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html)

[Variant Analysis](https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html)


# 0. Goal of this workflow

Objective: Compare the mitochondrial variants between mother and child (human)

We will learn: A.How to conduct "cleaning" of the data, B.How to map the sequence read to refeerence genome, C. How to call genetic variants.


# 1. Galaxy introduction
## 1-1. Register and login to Galaxy
Go to https://usegalaxy.no/ , https://usegalaxy.eu/ or https://usegalaxy.org/
Galaxy is a web platform with various software for genome analyses.
You should be able to log in with "Feide" information at galaxy.no (NMBU ID and password)

## 1-2. Play around with Galaxy

![](assets/BIO326-genome/Slide18.png)


![](assets/BIO326-genome/Slide19.png)
![](assets/BIO326-genome/Slide20.png)

![](assets/BIO326-genome/Slide21.png)



## 1-3. Review the genome analysis workflow

![](assets/BIO326-genome/Slide15.png)

# 2. Quality Control
## 2-0. Introduction

During sequencing, the nucleotide bases in a DNA or RNA sample (library) are determined by the sequencer. For each fragment in the library, a short sequence (=read) is generated,.

Modern sequencing technologies can generate a massive number of sequence reads in a single experiment. However, each instrument will generate different types and amount of errors, such as incorrect nucleotides being called. These wrongly called bases are due to the technical limitations.

Therefore, it is necessary to understand, identify and exclude error-types that may impact the interpretation of downstream analysis. Sequence quality control is therefore an essential first step in your analysis. Catching errors early saves time later on.


Objective:
Conduct quality control and trimming of sequence data from two individuals, mother and child 

We will learn:
How to conduct quality control with FastQC, how to interpret FastQC output, how to conduct trimming with fastp.

## 2-1. Prapere the data
Create a new history.
![](assets/BIO326-genome/Slide22.png)



Import the data set.

![](assets/BIO326-genome/Slide24.png)
Copy and paste the following URLs to import the data:

https://zenodo.org/record/1251112/files/raw_child-ds-1.fq

https://zenodo.org/record/1251112/files/raw_child-ds-2.fq

https://zenodo.org/record/1251112/files/raw_mother-ds-1.fq

https://zenodo.org/record/1251112/files/raw_mother-ds-2.fq


Rename the data for to manage them.
![](assets/BIO326-genome/Slide25.png)

Let's have a look...
![](assets/BIO326-genome/Slide26.png)

What is written in the FastQ files:
![](assets/BIO326-genome/Slide27.png)
Quality scores are recorded as "ASCII" symbol. The code is different in each platform.
For example, "B" is quality score "32", which means more than 30 = "Probability of incorrect base call" is less than "1 in 1000" (accuracy > 99.9%).

![](assets/BIO326-genome/Slide28.png)



## 2-2. Quality control with fastQC
[fastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) (go to the website for details)

![](assets/BIO326-genome/Slide29.png)



Let's see the result of quality check.
![](assets/BIO326-genome/Slide30.png)
Example results "Per base sequence quality":
X-axis is the base position, Y-axis is the read quality.
Green is good, yellow is okay, red is not good.
![](assets/BIO326-genome/Slide31.png)

Our results:
We see that the ending part ot our reads shows not good quality. -> Trimming.
![](assets/BIO326-genome/Slide32.png)
Per sequence quality score. 
X-axis is the quality score, Y-axis is the observed count.
We see that most of the reads show good quality, but still there are some low-quality reads.

![](assets/BIO326-genome/Slide33.png)

Sequence duplication levels.
There are some duplication (small peak) due to enrichment bias. These read will be removed in the further step.
![](assets/BIO326-genome/Slide34.png)


## 2-3. Trimming with fastp
[fastp](https://github.com/OpenGene/fastp) (go to the website for details)

![](assets/BIO326-genome/Slide35.png)
Let's see how many low-quality reads are removed:
![](assets/BIO326-genome/Slide36.png)


# 3. Mapping

Now, we are going to map reads to the reference genome (human genome)
FastQ format -> BAM format

![](assets/BIO326-genome/Slide37.png)

## 3-1. Mapping with BWA-MEM

![](assets/BIO326-genome/Slide38.png)
Taking too long?
Here are the bam files:
[mother](assets/BIO326-genome/mother.bam) and 
[child](assets/BIO326-genome/child.bam)

## 3-2. Merge the two BAM files
We will merge the mother and child BAM files into one file so that we can handle it easily.
![](assets/BIO326-genome/Slide39.png)

## 3-3. Remove the PCR duplicates (review 2-2.)

![](assets/BIO326-genome/Slide40.png)

## 3-4. Left-aligning insertion/deletion variants

![](assets/BIO326-genome/Slide42.png)
he same insertion/deletion can often be placed at multiple positions. To prevent this, we will conduct "Left-aligning" of insertion/deletion variants with "BamLeftAlign.
![](assets/BIO326-genome/Slide43.png)

## 3-5. BAM file filtering:

![](assets/BIO326-genome/Slide44.png)
 We will filter the bam file with quality and genomic region.
![](assets/BIO326-genome/Slide45.png)


# 4. Varitnt Calling

## 4-1. Calling variants with FreeBayes

Let's call variants from the BAM files and examine variants.
BAM format -> VCF (variant fall format)

![](assets/BIO326-genome/Slide46.png)
![](assets/BIO326-genome/Slide47.png)

![](assets/BIO326-genome/Slide48.png)

![](assets/BIO326-genome/Slide49.png)

Taking too long?
Here is the vcf file:
[VCF](assets/BIO326-genome/FreeBayes_mother_child.vcf)


## 4-2. Filtering variants
Filter the variant based on the reliability. There are various "biases" and artifacts that can happen.
![](assets/BIO326-genome/Slide50.png)

![](assets/BIO326-genome/Slide51.png)

![](assets/BIO326-genome/Slide52.png)

## 4-3. Reformatting the VCF file
Just reformat VCF file so that human eye can easily read it. 
![](assets/BIO326-genome/Slide53.png)
![](assets/BIO326-genome/Slide54.png)
There seem to be some version differences between Galaxy.no and Galaxy.org (2.2021)

![](assets/BIO326-genome/Slide55.png)


# 5. Examine the variants.

We see two variants at position 3243 and 5539.
Both are (Reference: A and Alternative:G)
![](assets/BIO326-genome/Slide56.png)
Let's take a close look on column five and six.
![](assets/BIO326-genome/Slide57.png)
![](assets/BIO326-genome/Slide58.png)
Q: Allele frequency of variants in Mother and Child

Posision 3243
Child : G = 0.67
Mother : G = ???

Posision 5538
Child : G = ???
Mother : G = ???

 


