---
title: "Bio326.genome"
author: "Marie Saitou"
date: "1/28/2021"
output: html_document
---




This workflow is based on Galaxy Training materials aimed for the two-hour coueses in [Bio326](https://www.nmbu.no/course/BIO326), NMBU 2021.

The workflow is based on the following materials but modified and updated for Galaxy.no 2021. 

[Quality Control](https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html)

[Mapping](https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html)

[Variant Analysis](https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html)


### 0. Goal of this workflow


Objective: Compare the mitochondrial variants between mother and child (human)
We will learn: A.How to conduct "cleaning" of the data, B.How to map the sequence read to refeerence genome, C. How to call genetic variants.


### 1. Galaxy introduction
#### 1-1. Register and login to Galaxy
Go to https://usegalaxy.no/ 
You should be able to log in with "Feide" information (NMBU ID and password)
Galaxy is a web platform with various software for genome analyses.


#### 1-2. Play around with Galaxy

![screen](assets/Bio326/Slide2.png)


![screen](assets/Bio326/Slide3.png)
![screen](assets/Bio326/Slide4.png)

![screen](assets/Bio326/Slide5.png)



#### 1-3. Review the workflow

![screen](assets/Bio326/Slide6.png)

### 2. Quality Control
#### 2-0. Introduction

During sequencing, the nucleotide bases in a DNA or RNA sample (library) are determined by the sequencer. For each fragment in the library, a short sequence (=read) is generated,.

Modern sequencing technologies can generate a massive number of sequence reads in a single experiment. However, each instrument will generate different types and amount of errors, such as incorrect nucleotides being called. These wrongly called bases are due to the technical limitations.

Therefore, it is necessary to understand, identify and exclude error-types that may impact the interpretation of downstream analysis. Sequence quality control is therefore an essential first step in your analysis. Catching errors early saves time later on.


##### Objective:
Conduct quality control and trimming of sequence data from two individuals, mother and child 
##### We will learn:
How to conduct quality control with FastQC, how to interpret FastQC output, how to conduct trimming with fastp.

#### 2-1. Prapere the data
Create a new history.
![screen](assets/Bio326/Slide7.png)



Import the data set.

![screen](assets/Bio326/Slide8.png)
Copy and paste the following URLs to import the data:
https://zenodo.org/record/1251112/files/raw_child-ds-1.fq
https://zenodo.org/record/1251112/files/raw_child-ds-2.fq
https://zenodo.org/record/1251112/files/raw_mother-ds-1.fq
https://zenodo.org/record/1251112/files/raw_mother-ds-2.fq

# Question: Why there are two sequence files per individual?




Let's have a look...
![screen](assets/Bio326/Slide9.png)
![screen](assets/Bio326/Slide10.png)

![screen](assets/Bio326/Slide11.png)

[fastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) (go to the website for details)

![screen](assets/Bio326/Slide12.png)

Taking too long? Download the data from here:
(Will be uploaded)
Let's have a look on the example below while running it... 

![screen](assets/Bio326/Slide13.png)
![screen](assets/Bio326/Slide14.png)

![screen](assets/Bio326/Slide15.png)
![screen](assets/Bio326/Slide16.png)
![screen](assets/Bio326/Slide17.png)

[fastp](https://github.com/OpenGene/fastp) (go to the website for details)

![screen](assets/Bio326/Slide18.png)

Mapping.

![screen](assets/Bio326/Slide21.png)
Taking too long? Download the data from here:

Here are the mapped data.


