---
title: "Genome Browser 2022"
author: "Marie Saitou"
date: "4/23/2022"
output: html_document
---

```{bash}

```

# 0 Introduction

[NCBI](https://www.ncbi.nlm.nih.gov/) 

National Center for Biotechnology Information. Good for primer design and litereture search.

[UCSC Genome Browser](https://genome.ucsc.edu/)

The grafical use intereface is light and helpful.
Most non-model species genomes are not up-to-date, but you can send them a request. 

[Ensembl](https://www.ensembl.org/index.html)

Most non-model species friendly. Good for evolutionary and functional genetics analysis.



# 1 Primer Design 

Mission: You want to investigate if TP53 gene is expressed in a particular tissue of pig and want to design a primer set for its cDNA.
 
## 1-1 Search for gene info at NCBI 

Go to:[NCBI](https://www.ncbi.nlm.nih.gov/labs/gquery/)
and search for "Sus scrofa  TP53".

![](assets/2022_browser/image-04-23-20.01.35.png)

Now you can see the gene information, exon-intron structure and gene/progein IDs.
![](assets/2022_browser/image-04-23-20.01.13.png)

Thin green lines are the introns, thick green boxes are exons.

Also when you scroll it down, you can see the expression pattern of this gene in various tissues.
![](assets/2022_browser/image-04-23-20.00.57.png)

RPKM is a unit for [gene expression](https://www.reneshbedre.com/blog/expression_units.html#:~:text=RPKM%20is%20a%20gene%20length,the%20expression%20of%20that%20gene.)

Furthere below, you can see relevant publications and expected function of this gene.

## 1-2 Primer blast - primer design

Now let's design a primer set to amplify the cDNA.
Put the mouse cursor on the gene image, and copy and note the refseq mRNA ID (NM_21824.3)

And go to:[NCBI primer blast](https://www.ncbi.nlm.nih.gov/tools/primer-blast/)

Put the transcript ID "NM_21824.3" in the PCR template box.

![](assets/2022_browser/image-04-23-20.03.30.png)


![](assets/2022_browser/image-04-23-20.03.25.png)

Specify "Primer mast span an exon-exon junction". --- to avoid false positive PCR amplification of genomic DNA and only observe mRNA by PCR. 

Also, Specify tne organism "pig". 


Wait for a while... 

Then the algorism gaves us the candidate primer sets. 
![](assets/2022_browser/image-04-23-20.09.07.png)

black bars are the exons and the blue thin lines are the primers and planned amplified region.

We can take a close look on each primer pair by clicking the primers.

![](assets/2022_browser/image-04-23-20.09.27.png)

We can now see the primer sequences, locations, melting tempreture, GC content, self complementarity, and the product length. It also tells us that the forward primer spans a exon-exon junction (location 1112+1113 th of the nucleotide)


## 1-3 Primer blast - primer reuse


Now you want to know if the primer sets can be used for other species. To do so, we can extract the target template sequence from the pig transcriptome data and compare it for those of other species.

![](assets/2022_browser/image-04-23-20.44.45.png)

When you put the cursor on primer1, it shows that this primer set will amplity the "1,100 - 1,750 th nucleotide" of the sequence from mRNA, refseq ID (NM_21824.3). Note the info so see if the PCR template can be observed in other species. 


![](assets/2022_browser/image-04-23-20.45.15.png)

Go to primer blast, and input NM_21824.3, and range "from 1100 - to 1750".
Add your favorite species in the "organism" space.



![](assets/2022_browser/image-04-23-20.46.59.png)



![](assets/2022_browser/image-04-23-20.47.52.png)


So probably we have to design new primer sets for these species...   


# 2 The effect of genetic variants

Mission: You found a genetic variant in your sequenced individual. You want to investigate the potential effect of the variants.


## 2-2 Variant effect predictor

Assume that you found the following deletion polymorphism at in a rabbit genome. 

12: 107,236,296-107,236,969


[Variant Effect Predictor](https://www.ensembl.org/Tools/VEP)
And input:

Species - rabbit (Oryctolagus_cuniculus)

Variant - 12 107236296 107236969 DEL + deletion1

the variant format, left to rignt ... chromosome, starting point, ending point, kind, strand, variant ID (you can name it as you like)
Variants.

You can also find various acceptable input formats [here](https://www.ensembl.org/info/docs/tools/vep/vep_formats.html)

![](assets/2022_browser/image-04-23-20.47.52.png)


## 2-2 What is this gene doing?

![](assets/2022_browser/image-04-23-20.47.52.png)

![](assets/2022_browser/image-04-23-20.47.52.png)

## 2-3 Comparative Analysis

![](assets/2022_browser/image-04-23-20.47.52.png)

![](assets/2022_browser/image-04-23-20.47.52.png)

Rabbit and Alpaca

# 3 the genomic architecture
## 3-1



## 3-2

genomic landscape 
genetic variants

## UCSC - check 






chr6:43,426,669-43,433,274

Repeats.


""
RepeatMasker Information

Name: (GGAT)n
```{bash, eval=FALSE}
>danRer11_rmsk_(GGAT)n range=chr6:43428241-43428346 5'pad=0 3'pad=0 strand=+ repeatMasking=none
GGATGGATGGATGGATGGATGGATGGATGGATGGATGGATGGATGGATGG
ATGGATGGATCGATGGAAGGATGGATGGATGGAAGGATGGATGGACAGAT
GGATGG
```


Variant.



