---
title: "Genome Browser 2022"
author: "Marie Saitou"
date: "4/23/2022"
output: html_document
---

# 0 Introduction - genome browsers

Aim: To get familiar with multiple genome browsers and explore potential use of their functions for your work.

[NCBI](https://www.ncbi.nlm.nih.gov/) 

National Center for Biotechnology Information. Good for primer design and litereture search.

[UCSC Genome Browser](https://genome-euro.ucsc.edu/cgi-bin/hgGateway?redirect=manual&source=genome.ucsc.edu)

The grafical use intereface is light and helpful.
Most non-model species genomes are not up-to-date, but you can send them a request. 

[Ensembl](https://www.ensembl.org/index.html)

Most non-model species friendly. Good for evolutionary and functional genetics analysis.



# 1 Primer Design 

Mission: You want to investigate if TP53 gene is expressed in a particular tissue of pig and want to design a primer set for its cDNA.
 
## 1-1 Search for gene info at NCBI 

Go to:[NCBI](https://www.ncbi.nlm.nih.gov/labs/gquery/)
and search for "Sus scrofa  TP53".

![](assets/2022_browser/image-04-23-20.01.35.png)

Now you can see the gene information, exon-intron structure and gene/progein IDs.
![](assets/2022_browser/image-04-23-20.01.13.png)

Thin green lines are the introns, thick green boxes are exons.

Also when you scroll it down, you can see the expression pattern of this gene in various tissues.
![](assets/2022_browser/image-04-23-20.00.57.png)

RPKM is a unit for [gene expression](https://www.reneshbedre.com/blog/expression_units.html#:~:text=RPKM%20is%20a%20gene%20length,the%20expression%20of%20that%20gene.)

Furthere below, you can see relevant publications and expected function of this gene.

## 1-2 Primer blast - primer design

Now let's design a primer set to amplify the cDNA.
Put the mouse cursor on the gene image, and copy and note the refseq mRNA ID (NM_21824.3)

And go to:[NCBI primer blast](https://www.ncbi.nlm.nih.gov/tools/primer-blast/)

Put the transcript ID "NM_21824.3" in the PCR template box.

![](assets/2022_browser/image-04-23-20.03.30.png)


![](assets/2022_browser/image-04-23-20.03.25.png)

Specify "Primer mast span an exon-exon junction". --- to avoid false positive PCR amplification of genomic DNA and only observe mRNA by PCR. 

Also, Specify tne organism "pig". 


Wait for a while... 

Then the algorism gaves us the candidate primer sets. 
![](assets/2022_browser/image-04-23-20.09.07.png)

black bars are the exons and the blue thin lines are the primers and planned amplified region.

We can take a close look on each primer pair by clicking the primers.

![](assets/2022_browser/image-04-23-20.09.27.png)

We can now see the primer sequences, locations, melting tempreture, GC content, self complementarity, and the product length. It also tells us that the forward primer spans a exon-exon junction (location 1112+1113 th of the nucleotide)


## 1-3 Primer blast - primer reuse


Now you want to know if the primer sets can be used for other species. To do so, we can extract the target template sequence from the pig transcriptome data and compare it for those of other species.

![](assets/2022_browser/image-04-23-20.44.45.png)

When you put the cursor on primer1, it shows that this primer set will amplity the "1,100 - 1,750 th nucleotide" of the sequence from mRNA, refseq ID (NM_21824.3). Note the info so see if the PCR template can be observed in other species. 


![](assets/2022_browser/image-04-23-20.45.15.png)

Go to primer blast, and input NM_21824.3, and range "from 1100 - to 1750".
Add your favorite species in the "organism" space.



![](assets/2022_browser/image-04-23-20.46.59.png)

There are cattle and sheep mRNA found. Dog may not have highly similar mRNA.

Let's look into the mRNA of sheep closely.


![](assets/2022_browser/image-04-23-20.47.52.png)

Query: 1100-1438 (pig TP53 mRNA) --- the original region was 1100-1750, but of the sequence, only - 1438 showed the high similarity with the sheep mRNA.


Subject: 1074-1436 (sheep TP53 mRNA)


[Blast Glossery](https://www.ncbi.nlm.nih.gov/books/NBK62051/)

Bit score ---  can be used to compare alignment scores (similarity) from different searches.
E-value (expect) --- the number of different alignments expected to occur in a database search by chance. 
Identities --- the number of identical nucleotids
Gaps --- the number of insertion/deletions in the alignment


The alignment indicates... the starting part is similar between sheep and pig mRNA, but 1439-1750 th bases of the pig mRNA did not show high similarity in sheep mRNA.

So probablly we have to design new primer sets for these species...   



# 2 The effect of genetic variants

Mission: You found a genetic variant in your sequenced individual. You want to investigate the potential effect of the variants.



## 2-2 Variant effect predictor

Assume that you found the following deletion polymorphism at in a rabbit genome. 

12: 107,236,296-107,236,969



[Variant Effect Predictor](https://www.ensembl.org/Tools/VEP)
Input the following information:

*Species - rabbit (Oryctolagus_cuniculus)*

*Variant - 12 107236296 107236969 DEL + deletion1*

the variant format, left to rignt ... chromosome, starting point, ending point, kind, strand, variant ID (you can name it as you like)
Variants.

![](assets/2022_browser/image-04-25-18.48.31.png)



You can also find various acceptable input formats [here](https://www.ensembl.org/info/docs/tools/vep/vep_formats.html)


![](assets/2022_browser/image-04-25-18.53.12.png)

The result shows that the deletion is affecting a gene, SLC35F1, by causing coding sequence variant, splice variant, feature truncation and intron variant.
You can modify the result table by clicking "show/hide columns"




## 2-2 What is this gene doing?

Click "ENSOCUG00000015307" to know more about this gene.

In the gene page, you can investigate its evolutionary history and potential function.



![](assets/2022_browser/image-04-25-19.10.05.png)

Let's explore "gene gain/loss tree" "Ontologies" and "Phenotypes"

Gene gain/loss tree

![](assets/2022_browser/image-04-25-19.11.48.png){width=80%}

Most of mammalian species have two copies of this gene.


![](assets/2022_browser/image-04-25-19.12.04.png){width=80%}

See what is happening in fish!


Gene Ontology 
![](assets/2022_browser/image-04-25-19.12.34.png)
transmembrane transtport


![](assets/2022_browser/image-04-25-19.13.13.png)

Phenotypes
There is no phenotype information for this gene in rabbits, but there are some reports in mouse and rat, such as leukocyte cell number etc.


## 2-3 Comparative Analysis


Go back to the gene page, and click *Orthologues* in the left pane.

Find Alpaca... and let's compare this gene in Rabbit and Alpaca.

![](assets/2022_browser/image-04-25-19.15.59.png)
*"Compare regions"* 

![](assets/2022_browser/image-04-25-19.15.44.png)
In both rabbit (top) and alpaca (bottom), seemingly, SLC35F1 is on the upstream of the CEF85L gene. 

![](assets/2022_browser/image-04-25-19.15.59.png)

*"View Sequence Alignments -- Protein"* 

![](assets/2022_browser/image-04-25-19.17.24.png)

This is the protein alignment of this gene in the two species.
Seemingly, these two sequences are very conserved (similar) except for the first 60 amino acids. * indicates the identical sequence.



# 3 The genomic architecture

Mission: Investigate the genomic arfhitecture of a particular region in the zebrafish genome

Go to [UCSC Genome Browser](https://genome-euro.ucsc.edu/cgi-bin/hgGateway?redirect=manual&source=genome.ucsc.edu)

Select Zebrafish Genome, and search for "chr6:43,426,669-43,433,274".



Now we see a gene structure at this region from multiple data sources - Thin lines are the introns, thick boxes are exons.

![](assets/2022_browser/image-04-25-21.05.35.png)



![](assets/2022_browser/image-04-25-21.05.45.png)

If you click RepeatMasker, you can see various different tipes of repeats in this reagion. What does it mean...? Click the left black part of "simple".

![](assets/2022_browser/image-04-25-21.06.35.png){width=80%}


RepeatMasker Information

*View table schema* (<-click this in your browser)

```{bash, eval=FALSE}
>danRer11_rmsk_(GGAT)n range=chr6:43428241-43428346 5'pad=0 3'pad=0 strand=+ repeatMasking=none
GGATGGATGGATGGATGGATGGATGGATGGATGGATGGATGGATGGATGG
ATGGATGGATCGATGGAAGGATGGATGGATGGAAGGATGGATGGACAGAT
GGATGG
```

So, this means that in this region (chr6:43428241-43428346), there is repeats of "GGAT" sequences. This may happen during evolution - mostly by random or mechanistic mutations, and some are due to insertion og viral genetic sequences. Some of them could help a gene to gain a new function, and some can break a gene.



If you click EVA SNP Release3, you can see a lot of known variants in this region.
rsXXX are the IDs of genetic variants.


![](assets/2022_browser/image-04-25-21.07.44.png)

![](assets/2022_browser/image-04-25-19.23.22.png)


If you click "rs179383132", you can see more information about this variant.

![](assets/2022_browser/image-04-25-21.07.54.png)


