---
title: "RNAseq tutorial for lab people"
author: "Marie Saitou"
date: "4/21/2022"
output: html_document
---

## Lecture slides (currently dummy)
Marie:  Overview of the RNAseq. What can RNAseq tell us/What should we do when we write a manuscript. [slides](assets/BIO326-genome/BIO326-genome.pdf)
Matthew: RNAseq lab part mechanism, experimental design etc.[slides](assets/BIO326-genome/BIO326-genome.pdf)
Domniki: How to upload data to ENA. [slides](assets/BIO326-genome/BIO326-genome.pdf)


# 0. Goal of this workflow

Objective: Conduct comparative RNA sequencing analyis between fish brain and liver datasets

We will learn: 
A. How to conduct "cleaning" of the RNAseq data. 
B. How to quantify the gene expression in each sample.
C. How to conduct downstream analysis to gain biological insights.


# 1. Galaxy introduction
Useful materials:
https://training.galaxyproject.org/training-material/topics/introduction/


## 1-1. Register and login to Galaxy
Go to https://usegalaxy.no/ ,
Galaxy is a web platform with various software for genome analyses.
You should be able to log in with "Feide" information at galaxy.no (NMBU ID and password)
If you are not NMBU employee, you can use these for free: https://usegalaxy.eu/ or https://usegalaxy.org/

## 1-2. Play around with Galaxy


![](assets/BIO326-genome/Slide18.png)
![](assets/BIO326-genome/Slide19.png)
![](assets/BIO326-genome/Slide20.png)
![](assets/BIO326-genome/Slide21.png)

## 1-3. Review the RNAseq analysis workflow

On Galaxy
- Get RNAseq data (paired end, liver and brain, three samples each) 
- Get reference transcriptome: Salmo salar version 3.1
- fastp: trimming low-quality reads
- Kallisto: quantify gene expression from the RNAseq data and reference
- tximport: summarize transcripts into genes

On iDEP
- Upload gene expression table
- Quality check
- Plot genes of interest 
- Differentially expressed gene
- Gene ontology analysis


# 2. Get the data

There are three major public repository for genomics data ENA (Europe), NCBI (America) DDBJ (Japan) -- which are regularly synchronized. 


This time, we will analyze brain and liver transcriptome datasets from the following study:
"Multi-tissue transcriptome profiling of North American derived Atlantic salmon"
Paper: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6146974/
Dataset: https://www.ebi.ac.uk/ena/browser/view/PRJNA470665





# 3. Quality Control
## 2-0. Introduction
https://www.ensembl.org/Salmo_salar/Info/Index

http://ftp.ensembl.org/pub/release-106/fasta/salmo_salar/cdna/Salmo_salar.Ssal_v3.1.cdna.all.fa.gz



Get the reference
```{bash, eval=FALSE}


```


Get the RNAseq data
At it takes hours to analyze the real (gigabytes) data, I made a miniature datasets for each samples. 
```{bash, eval=FALSE}


```

GSM3138423	brain_rep1
GSM3138424	brain_rep2
GSM3138425	brain_rep3
GSM3138435	liver_rep1
GSM3138436	liver_rep2
GSM3138437	liver_rep3



```{bash, eval=FALSE}
# Real brain samples (two files per sample)


# Real liver samoles (two files per sample)


```


<details>
  <summary> **If you want to try with real data, click here**</summary>      
```{bash, eval=FALSE}
# Real brain samples (two files per sample)
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/005/SRR7139945/SRR7139945_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/005/SRR7139945/SRR7139945_2.fastq.gz

ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/007/SRR7139947/SRR7139947_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/007/SRR7139947/SRR7139947_2.fastq.gz

ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/000/SRR7139950/SRR7139950_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/000/SRR7139950/SRR7139950_2.fastq.gz

# Real liver samoles (two files per sample)
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/009/SRR7139969/SRR7139969_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/009/SRR7139969/SRR7139969_2.fastq.gz

ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/001/SRR7139971/SRR7139971_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/001/SRR7139971/SRR7139971_2.fastq.gz

ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/003/SRR7139973/SRR7139973_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/003/SRR7139973/SRR7139973_2.fastq.gz

```
</details> 

<details>
  <summary> **A note to make tutorial files**</summary>      
```{bash, eval=FALSE}
for file in *.fastq.gz; do zcat < $file | head -100000 | gzip > {$file}_mini.fastq.gz; done
```
</details> 


## Galaxy

# https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE114247

Quality check



Trimming of bad quality data and adaptors
Fastp

Kallisto

![](assets/BIO326-genome/Slide18.png)

![](assets/RNAseq_for_lab_2022/pairing.png)


<details>
  <summary> **What fastp does (from Galaxy page)**</summary>      
```{bash, eval=FALSE}
fastp is a tool designed to provide fast all-in-one preprocessing for FASTQ files. 

Features

- Filter out bad reads (too low quality, too short, or too many N...)
- Cut low quality bases for per read in its 5' and 3' by evaluating the mean quality from a sliding window (like Trimmomatic but faster)
Trim all reads in front and tail
- Cut adapters. Adapter sequences can be automatically detected, which means you don't have to input the adapter sequences to trim them.
- Correct mismatched base pairs in overlapped regions of paired end reads, if one base is with high quality while the other is with ultra-low quality
- Trim polyG in 3' ends, which is commonly seen in NovaSeq/NextSeq data. Trim polyX in 3' ends to remove unwanted polyX tailing (i.e. polyA tailing for mRNA-Seq data)
- Preprocess unique molecular identifer (UMI) enabled data, shift UMI to sequence name
- Report JSON format result for further interpreting
- Visualize quality control and filtering results on a single HTML page (like FASTQC but faster and more informative)
- Split the output to multiple files (0001.R1.gz, 0002.R1.gz...) to support parallel processing. Two modes can be used, limiting the total split file number, or limitting the lines of each split file (Not enabled in this Galaxy tool)
- Support long reads (data from PacBio / Nanopore devices)

```
</details> 





<details>
  <summary> **What Kallisto does (from Galaxy page)**</summary>      
```{bash, eval=FALSE}

kallisto is a program for quantifying abundances of transcripts from RNA-Seq data, or more generally of target sequences using high-throughput sequencing reads. It is based on the novel idea of pseudoalignment for rapidly determining the compatibility of reads with targets, without the need for alignment. On benchmarks with standard RNA-Seq data, kallisto can quantify 30 million human reads in less than 3 minutes on a Mac desktop computer using only the read sequences and a transcriptome index that itself takes less than 10 minutes to build. Pseudoalignment of reads preserves the key information needed for quantification, and kallisto is therefore not only fast, but also as accurate as existing quantification tools. In fact, because the pseudoalignment procedure is robust to errors in the reads, in many benchmarks kallisto significantly outperforms existing tools.

```
</details> 

![](assets/RNAseq_for_lab_2022/PCA.png)

![](assets/RNAseq_for_lab_2022/Kallisto.png)

If it is taking too long, here is the output file.
sample * gene expression table.




## IDEP



![](assets/RNAseq_for_lab_2022/PCA.png)
![](assets/RNAseq_for_lab_2022/KEGG.png)

![](assets/RNAseq_for_lab_2022/Kallisto.png)








