---
title: "RNAseq tutorial without coding"
author: "Marie Saitou"
date: "5/26/2022"
output: html_document
---

## Lecture slides 
Marie:  Overview of the RNAseq. What can RNAseq tell us/What should we do when we write a manuscript. [slides](https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/RNAseq_Jun_2022.pdf)

Matthew: RNAseq lab part mechanism, experimental design etc.
(coming soon)

Domniki: How to upload data to ENA. 
(coming soon)

# 0. Goal of this workflow

Objective: Conduct comparative RNA sequencing analyis between fish brain and liver datasets

We will learn: 

A. How to conduct "cleaning" of the RNAseq data. 

B. How to quantify the gene expression in each sample.

C. How to conduct downstream analysis to gain biological insights.


# 1. Galaxy introduction
Useful materials:

https://training.galaxyproject.org/training-material/topics/introduction/


## 1-1. Register and login to Galaxy
Go to https://usegalaxy.no/ ,

Galaxy is a web platform with various software for genome analyses.

You should be able to log in with "Feide" information at galaxy.no (NMBU ID and password)

If you are not NMBU employee, you can use these for free: https://usegalaxy.eu/ or https://usegalaxy.org/


## 1-2. Play around with Galaxy


![](assets/BIO326-genome/Slide18.png)
![](assets/BIO326-genome/Slide19.png)
![](assets/BIO326-genome/Slide20.png)
![](assets/BIO326-genome/Slide21.png)

## 1-3. Review the RNAseq analysis workflow
```{bash, eval=FALSE}
On Galaxy
- Get RNAseq data (paired end, liver and brain, three samples each) 
- Get reference transcriptome: Salmo salar version 3.1
- fastp: trimming low-quality reads
- Kallisto: quantify gene expression from the RNAseq data and reference
- tximport: summarize transcripts into genes

On iDEP
- Upload gene expression table
- Quality check
- Plot genes of interest 
- Differentially expressed gene
- Gene ontology analysis
```


# 2. Get the data

There are three major public repository for genomics data ENA (Europe), NCBI (America) DDBJ (Japan) -- which are regularly synchronized. 


This time, we will analyze brain and liver transcriptome datasets from the following study:
"Multi-tissue transcriptome profiling of North American derived Atlantic salmon"

Paper: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6146974/

Dataset: https://www.ebi.ac.uk/ena/browser/view/PRJNA470665


![](assets/RNAseq_for_lab_2022/get_data.png)
Click "upload data" on the left top - and 

Select Paste/Fetch data and copy/paste the URL of the datasets.

Choose local files --- upload data from your computer.

Choose remote files --- if your Galaxy storage (200Gb in total) is getting full, you can store large files at the NeLS Storage and access them here. 

To transfer files to the NeLS Storage, click Send Data - Export datasets in the left menu and select NeLS Storage as the destination.
![](assets/RNAseq_for_lab_2022/storage.png)



## 2-1. Get the liver and brain RNAseq data

As it takes hours to analyze the real (gigabytes) data, I made a miniature datasets for each samples. 
```{bash, eval=FALSE}

https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139945_1_mini.fastq.gz
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139945_2_mini.fastq.gz
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139950_1_mini.fastq.gz
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139950_2_mini.fastq.gz
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139969_1_mini.fastq.gz
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139969_2_mini.fastq.gz
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139971_1_mini.fastq.gz
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139971_2_mini.fastq.gz
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139973_1_mini.fastq.gz
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139973_2_mini.fastq.gz
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139947_1_mini.fastq.gz
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139947_2_mini.fastq.gz

```


## 2-2. Get the reference
```{bash, eval=FALSE}
http://ftp.ensembl.org/pub/release-106/fasta/salmo_salar/cdna/Salmo_salar.Ssal_v3.1.cdna.all.fa.gz

## New! Hien newly made a chromosomes + mitochondrial genome version

http://arken.nmbu.no/~lagr/share/Ssal.withMT/ensembl/Salmo_salar.Ssal_v3.1.106.withMT.gff3.gz
```
(If it does not work well, download it on your computer and upload it to Galaxy afterword)




If you need reference sequences of other species for your study, go to  https://www.ensembl.org/index.html

and "select a species" in the all genomes tub.

Go to "Gene annotation", Download FASTA files and cDNA sequence.





<details>
  <summary> **If you want to try with real data later, click here**</summary>      
```{bash, eval=FALSE}
# Real brain samples (two files per sample)
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/005/SRR7139945/SRR7139945_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/005/SRR7139945/SRR7139945_2.fastq.gz

ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/007/SRR7139947/SRR7139947_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/007/SRR7139947/SRR7139947_2.fastq.gz

ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/000/SRR7139950/SRR7139950_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/000/SRR7139950/SRR7139950_2.fastq.gz

# Real liver samoles (two files per sample)
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/009/SRR7139969/SRR7139969_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/009/SRR7139969/SRR7139969_2.fastq.gz

ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/001/SRR7139971/SRR7139971_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/001/SRR7139971/SRR7139971_2.fastq.gz

ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/003/SRR7139973/SRR7139973_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR713/003/SRR7139973/SRR7139973_2.fastq.gz

```
</details> 

<details>
  <summary> Marie's note to make tutorial files</summary>      
```{bash, eval=FALSE}
for file in *.fastq.gz; do zcat < $file | head -100000 | gzip > {$file}_mini.fastq.gz; done
```
</details> 


# 3. Quality Control


## 3-1. Pairing of the paired read data sets

[What is Paired-End Sequencing?](https://www.illumina.com/science/technology/next-generation-sequencing/plan-experiments/paired-end-vs-single-read.html)


As each sample has two files (paired-end), we make a "collection of paired datasets" to handle many files conveniently.

![](assets/RNAseq_for_lab_2022/pairing.png)

## 3-2. Trimming of bad quality data and adaptors with fastp

<details>
  <summary> **What fastp does (from Galaxy page)**</summary>      

fastp is a tool designed to provide fast all-in-one preprocessing for FASTQ files. 

Features

- Filter out bad reads (too low quality, too short, or too many N...)
- Cut low quality bases for per read in its 5' and 3' by evaluating the mean quality from a sliding window (like Trimmomatic but faster)
Trim all reads in front and tail
- Cut adapters. Adapter sequences can be automatically detected, which means you don't have to input the adapter sequences to trim them.
- Correct mismatched base pairs in overlapped regions of paired end reads, if one base is with high quality while the other is with ultra-low quality
- Trim polyG in 3' ends, which is commonly seen in NovaSeq/NextSeq data. Trim polyX in 3' ends to remove unwanted polyX tailing (i.e. polyA tailing for mRNA-Seq data)
- Preprocess unique molecular identifer (UMI) enabled data, shift UMI to sequence name
- Report JSON format result for further interpreting
- Visualize quality control and filtering results on a single HTML page (like FASTQC but faster and more informative)
- Split the output to multiple files (0001.R1.gz, 0002.R1.gz...) to support parallel processing. Two modes can be used, limiting the total split file number, or limitting the lines of each split file (Not enabled in this Galaxy tool)
- Support long reads (data from PacBio / Nanopore devices)


</details> 

Search for a tool "fastp" in the left window (Tools) and put the input files we gathered in the 3-1.

![](assets/RNAseq_for_lab_2022/fastp.png)

Let's take a look on how much/what kind of reads were removed.

![](assets/RNAseq_for_lab_2022/fastp_result.png)


## 3-3. Quantify the gene expression on the cleaned reads with Kallisto


<details>
  <summary> **What Kallisto does (from Galaxy page)**</summary>      


kallisto is a program for quantifying abundances of transcripts from RNA-Seq data, or more generally of target sequences using high-throughput sequencing reads. It is based on the novel idea of pseudoalignment for rapidly determining the compatibility of reads with targets, without the need for alignment. On benchmarks with standard RNA-Seq data, kallisto can quantify 30 million human reads in less than 3 minutes on a Mac desktop computer using only the read sequences and a transcriptome index that itself takes less than 10 minutes to build. Pseudoalignment of reads preserves the key information needed for quantification, and kallisto is therefore not only fast, but also as accurate as existing quantification tools. In fact, because the pseudoalignment procedure is robust to errors in the reads, in many benchmarks kallisto significantly outperforms existing tools.

</details> 


![](assets/RNAseq_for_lab_2022/Kallisto.png)

If it is taking too long, here are the output files.
The output files are six in total, transcript * transcript counts tables.

```{bash, eval=FALSE}
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139945_mini.Kallisto.tabular
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139947_mini.Kallisto.tabular
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139969_mini.Kallisto.tabular
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139971_mini.Kallisto.tabular
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139973_mini.Kallisto.tabular
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/SRR7139950_mini.Kallisto.tabular
```



## 3-4. Summarize the transcripts to genes with tximport

Upload a transcript-gene table to Galaxy.

```{bash, eval=FALSE}
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/transcript_gene_ssav3.txt
```


(This table can be obtained at [Ensemnbl BiomaRt](https://www.ensembl.org/biomart/martview/3ed7f9aeecadada91863b573c33aefdc) for many species.)


![](assets/RNAseq_for_lab_2022/tximport.png)




# 4. Downstream analysis with iDEP
## 4-1. Data upload
Download the results of the entire RNAseq analysis.

```{bash, eval=FALSE}
https://github.com/mariesaitou/Bio326/blob/master/docs/assets/RNAseq_for_lab_2022/materials/rnaseq_liver_brain.txt
```

And go to iDEP

http://bioinformatics.sdstate.edu/idep/

Upload the gene expression matrix and iDEP automatically runs (it detect the species as well)

![](assets/RNAseq_for_lab_2022/input_iDEP.png){width=60%}


## 4-2. Quality check "Pre-Process"

Keep genes with minimal counts per million (CPM) in at least n libraries:

Min. CPM: 1, 2 libraries

VST: variance stabilizing transform

(I prefer to make it a bit stricter than default setting).


This is a density plot of normalized transcriptome of each sample. Brain and liver look slightly different, which makes sense. If some samples look obviously different from others (such as the expression is too high/low), it may be better to remove those samples.

![](assets/RNAseq_for_lab_2022/density.png)


Transcriptome comparison between brain1 and brain2 samples: high correlation.

![](assets/RNAseq_for_lab_2022/correl_brain.png){width=60%}

Transcriptome comparison between brain and liver samples: not very high correlation.

![](assets/RNAseq_for_lab_2022/correl_brain_liver.png){width=60%}

## 4-3.Principal component analysis (PCA)

Visualize overall trend of all the gene expression. Each dot is each sample, colored by tissue.

X axis - Principal component 1 (combination of certain gene expressions) explains 88% of the transcriptome variation across samples (mostly the difference between brain and liver). 

Y axis - Principal component 2 (combination of certain gene expressions) explains 6% of the transcriptome variation across samples (mostly the difference among brain). 

If one sample is obviously distant from others (=shows unusual expression patteren), it may be better to remove that sample from the analysis.

![](assets/RNAseq_for_lab_2022/PCA.png)



## 4-4. Visualize the expression of your favorite genes 


"Pre-Process" -> "Plot one or more genes"

Type your favorite genes... 

![](assets/RNAseq_for_lab_2022/barplot.png)

## 4-5. Differentially expressed genes (DEG)

Go to DEG1 tab

"DESeq2"

False discovery ratio cutoff: 0.05

Minimum fold change: 1.5

(You can download the list of deferentially expressed genes)


Go to DEG2 tab

liver-brain comparison

downregulated: more expression in brain (metabolic genes),

unregulated: more expression in liver (signaling genes)

Volcano Plot, check "Show interactive version w/ gene symbols".

X-axis: fold change, Y-axis: false discovery ratio (statistical significance of the difference)


![](assets/RNAseq_for_lab_2022/Volcano.png)

"Enrichment analysis for DEGs:"

Gene Ontology Analysis of differentially expressed genes
![](assets/RNAseq_for_lab_2022/GO.png)
Also, do play around with Enrichment tree, Network etc. 



KEGG Analysis - if you have a particular pathway of interest.


![](assets/RNAseq_for_lab_2022/KEGG.png)


Others. 

Genome: if you are interested in the distributions of the differentially expressed genes in the genome.

K-Means: if you have three or more conditions, "K-means" can classify the gene expression pattern more than up/down.



Enjoy!




